### DISCUSSION TITLE
Access to function behind "copy to clipboard"

### DISCUSSION THREAD

**Discussion by 3goats** [squidfunk/mkdocs-material#4005] [General]
I want to add some custom javascript that provides an "execute" or "run" button alongside the "copy to clipboard" button for code blocks. Is the "copy to clipboard" function accessible somehow so that I can refuse it grab the code block and do something with it ? 

Any pointers as to how to achieve this would be much appreciated.

**Comment by squidfunk** [comment]
The copy-to-clipboard button is mounted at this location:

https://github.com/squidfunk/mkdocs-material/blob/cfe83ccbbaf8ac33d39187b289ec61cd0a349cb5/src/assets/javascripts/components/content/code/_/index.ts#L170-L178

You can fork the theme and add other buttons to your liking. Note that you have to go through the [build process](https://squidfunk.github.io/mkdocs-material/customization/#theme-development).
